docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,169 @@
+:EP: 38
+:Title: Action Pacing
+:Authors:
+    - David Ramirez <davramir@fiu.edu>;
+:Issued Date: 2024-05-16
+:Status: Pending
+:Type: Standards Track
+
+*********************
+EP038 - Action Pacing
+*********************
+
+########
+Abstract
+########
+
+This document is intended on a guide for both NApp Developers
+and Network Operators for how to implement and use Pacing within Kytos.
+Pacing is intended as a mechanism to control
+how often certain actions will be executed.
+This differs from rate limiting, as the action is not cancelled,
+and instead the caller is temporarily paused until the action
+is allowed to be executed again.
+For an action to be paced, it needs to be instrumented
+by developers to support pacing.
+
+##########
+Motivation
+##########
+
+The purpose of adding in pacing to Kytos is
+help maintain system stability by preventing
+excessive throughput.
+It could occur that sending too many flow mods
+or too many DB requests may result in
+connection failures, and therefore system
+instability.
+To prevent such scenarios,
+pacing of actions which are sensitive to high
+throuhput should be implemented.
+
+#############
+Specification
+#############
+
+This section describes how to set the pace of actions,
+intended for network operators,
+and how to add pacing to an action,
+intended for developers.
+
+Setting A Pace
+==============
+
+For network operators,
+setting the pace of an action can be done through
+a NApp's ``action_paces`` setting.
+The ``action_paces`` setting is a dictionary,
+with keys representing the action to be paced,
+and values being the rate to pace them to.
+The pacing value is a string formatted as
+``{max_concurrent}/{time_period}{time_unit}``,
```

Comment:
Which `time_unit`s will be supported? Let's document them.

Reply:
Time units now documented.